ghost/core/test/e2e-server/jobs/update-check.test.js

```diff
@@ -0,0 +1,54 @@
+const assert = require('assert');
+const http = require('http');
+const path = require('path');
+
+const models = require('../../../core/server/models');
+
+models.init();
+
+const jobService = require('../../../core/server/services/jobs/job-service');
+
+const JOB_NAME = 'update-check';
+const JOB_PATH = path.resolve(__dirname, '../../../core/server/run-update-check.js');
+
+describe('Run Update Check', function () {
+    let mockUpdateServer;
+
+    afterEach(function () {
+        if (mockUpdateServer) {
+            mockUpdateServer.close();
+        }
+    });
+
+    it('successfully executes the update checker', async function () {
+        let mockUpdateServerRequestCount = 0;
+
+        // Initialise mock update server
+        mockUpdateServer = http.createServer((req, res) => {
```

Comment:
I'd substitute this with `nock`, which we use everywhere else ([example](https://github.com/TryGhost/Ghost/blob/b58dac262ec7075daa3b3d948cc4860d171bb20a/ghost/members-api/test/unit/lib/geolocation.test.js#L32)) to mock external HTTP dependencies.

Reply:
We wouldn't be able to use nock here because the job manager executes the job in a separate process

Reply:
Comment added in https://github.com/TryGhost/Ghost/pull/16840/commits/154de6ff46a02e4039650eb1b8d2b98cf9cc096b